lib/icingadb/icingadb.ti

```diff
@@ -12,36 +12,37 @@ class IcingaDB : ConfigObject
 {
 	activation_priority 100;
 
-	[config] String host {
+	[config, no_user_modify] String host {
 		default {{{ return "127.0.0.1"; }}}
 	};
-	[config] int port {
+	[config, no_user_modify] int port {
 		default {{{ return 6380; }}}
 	};
-	[config] String path;
+	[config, no_user_modify] String path;
+	[config, no_user_modify] String username;
 	[config, no_user_view, no_user_modify] String password;
-	[config] int db_index;
+	[config, no_user_modify] int db_index;
 
-	[config] bool enable_tls {
+	[config, no_user_modify] bool enable_tls {
 		default {{{ return false; }}}
 	};
 
-	[config] bool insecure_noverify {
+	[config, no_user_modify] bool insecure_noverify {
 		default {{{ return false; }}}
 	};
 
-	[config] String cert_path;
-	[config] String key_path;
-	[config] String ca_path;
-	[config] String crl_path;
-	[config] String cipher_list {
+	[config, no_user_modify] String cert_path;
+	[config, no_user_modify] String key_path;
+	[config, no_user_modify] String ca_path;
+	[config, no_user_modify] String crl_path;
+	[config, no_user_modify] String cipher_list {
 		default {{{ return DEFAULT_TLS_CIPHERS; }}}
 	};
-	[config] String tls_protocolmin {
+	[config, no_user_modify] String tls_protocolmin {
 		default {{{ return DEFAULT_TLS_PROTOCOLMIN; }}}
 	};
 
-	[config] double connect_timeout {
+	[config, no_user_modify] double connect_timeout {
```

Comment:
Indeed, elasticsearchwriter and influxdb*writer handle API changes well.

In contrast, the others still use the old connection as long as it's still there. Shall I let Icinga DB just behave like this instead of forbidding changes? (I.e. RedisConnection#Connect() takes all data from IcingaDB every time.)

Reply:
> Shall I let Icinga DB just behave like this instead of forbidding changes? (I.e. RedisConnection#Connect() takes all data from IcingaDB every time.)

If you want to rewrite the whole Redis connection thing all over again now just to allow to do some fancy stuff that nobody ever uses or asked for, then keep me out of this, but IMHO that's going to be just a wasted time instead of focusing on issues/bugs that really matter.